src/PIL/ImageShow.py

```diff
@@ -234,150 +235,80 @@ def show_file(self, path=None, **options):
                 path = options.pop("file")
             else:
                 raise TypeError("Missing required argument: 'path'")
-        subprocess.Popen(["xdg-open", path])
-        self._remove_path_after_delay(path)
+
+        path = quote(path)
+        command = self.get_command(path, **options)
+
+        kwargs = {
+            "stdout": subprocess.PIPE,
+        }
+
+        th = threading.Thread(
+            target=subprocess.run, args=(command.split(),), kwargs=kwargs, name=path
+        )
+        self.opened_images.append(th.name)
+        th.start()
+        th_monitor = threading.Thread(target=self.thread_monitor, args=(th,))
+        th_monitor.start()
         return 1
 
 
-class DisplayViewer(UnixViewer):
+class XDGViewer(UnixViewer):
     """
-    The ImageMagick ``display`` command.
-    This viewer supports the ``title`` parameter.
+    The freedesktop.org ``xdg-open`` command.
     """
 
-    def get_command_ex(self, file, title=None, **options):
-        command = executable = "display"
-        if title:
-            command += f" -name {quote(title)}"
+    def get_command_ex(self, file, **options):
+        executable = "xdg-open"
+        command = f"xdg-open {quote(file)}"
         return command, executable
 
-    def show_file(self, path=None, **options):
-        """
-        Display given file.
 
-        Before Pillow 9.1.0, the first argument was ``file``. This is now deprecated,
-        and ``path`` should be used instead.
-        """
-        if path is None:
-            if "file" in options:
-                warnings.warn(
-                    "The 'file' argument is deprecated and will be removed in Pillow "
-                    "10 (2023-07-01). Use 'path' instead.",
-                    DeprecationWarning,
-                )
-                path = options.pop("file")
-            else:
-                raise TypeError("Missing required argument: 'path'")
-        args = ["display"]
-        if "title" in options:
-            args += ["-name", options["title"]]
-        args.append(path)
+class DisplayViewer(UnixViewer):
+    """
+    The ImageMagick ``display`` command. This viewer supports the ``title`` parameter.
+    """
 
-        subprocess.Popen(args)
-        os.remove(path)
-        return 1
+    def get_command_ex(self, file, title=None, **options):
+        executable = "display"
+        command = f"display {quote(file)}"
+        if title:
+            command = f"display -title {quote(title)} {quote(file)}"
+        return command, executable
 
 
 class GmDisplayViewer(UnixViewer):
     """The GraphicsMagick ``gm display`` command."""
 
     def get_command_ex(self, file, **options):
         executable = "gm"
-        command = "gm display"
+        command = f"gm display {quote(file)}"
```

Comment:
As I've said before, I consider this to be a backwards incompatible change.

Reply:
I think that it is more convenient to keep a viewer opening shell command at one place. As excuse I will add, that get_command_ex() is a helper function, and the interface of the function was preserved.